examples/interaction_box.py

```diff
@@ -0,0 +1,51 @@
+"""
+Demonstrate interaction box on points layer
+"""
+
+from skimage import data
+import napari
+import numpy as np
+from napari.layers.points._points_utils import points_in_box
+
+
+def on_selection_box_drag(event):
+    sel_i = points_in_box(event.box,viewer.layers.selection.active._view_data,viewer.layers.selection.active._view_size)
+    viewer.layers.selection.active.selected_data = sel_i
+    
+def on_selection_box_final(event):
+    sel_i = points_in_box(event.box,viewer.layers.selection.active._view_data,viewer.layers.selection.active._view_size)
+    viewer.layers.selection.active.selected_data = sel_i
+    viewer.layers.selection.active._interaction_box.points = np.array([viewer.layers.selection.active._view_data[i] for i in sel_i])
```

Comment:
```suggestion
    viewer.layers.selection.active._interaction_box.points = viewer.layers.selection.active._view_data[sel_i]
```

If the interaction box needs a copy of the data, you can wrap the RHS in `np.array(...)`.

There are some other similar changes in this file, so if this looks good, please make those too.